vj4/model/adaptor/contest.py

```diff
@@ -160,15 +168,27 @@ def get_multi_status(*, fields=None, **kwargs):
   if 'attend' not in tsdoc or not tsdoc['attend']:
     raise error.ContestNotAttendedError(domain_id, tid, uid)
 
-  # Sort and uniquify journal of the contest status document, by rid.
-  key_func = lambda j: j['rid']
-  journal = [list(g)[-1]
-             for _, g in itertools.groupby(sorted(tsdoc['journal'], key=key_func), key=key_func)]
+  journal = _get_status_journal(tsdoc)
   stats = RULES[tdoc['rule']].stat_func(tdoc, journal)
   tsdoc = await document.rev_set_status(domain_id, document.TYPE_CONTEST, tid, uid, tsdoc['rev'],
                                         journal=journal, **stats)
   return tsdoc
 
 
+@argmethod.wrap
+async def recalc_contest_status(domain_id: str, tid: objectid.ObjectId):
```

Comment:
Moving to `job` would be better?

Reply:
it is not triggered periodically?

Reply:
Jobs are not triggered periodically.

Reply:
`recalc_contest_status` don't need to be run manually